test/common/tasks/LargeFileUploadTask.ts

```diff
@@ -147,90 +133,244 @@ describe("LargeFileUploadTask.ts", () => {
+				};
+
+				const progressCallBack: Progress = {
+					progress,
+					completed,
+					failure,
+					extraCallbackParams: true,
+				};
+
+				const optionsWithProgress: LargeFileUploadTaskOptions = {
+					rangeSize,
+					progressCallBack,
+				};
+
+				const emptyBody = {};
+				const uploadTask = new LargeFileUploadTask(getClient(), fileObj, uploadSession, optionsWithProgress);
+				const status201 = {
+					status: 201,
+					statusText: "OK",
+					headers: {
+						"Content-Type": "application/json",
+						location,
+					},
+				};
+				const rawResponse = new Response(JSON.stringify(emptyBody), status201);
+
+				const moq = sinon.mock(uploadTask);
+				moq.expects("uploadSliceGetRawResponse").resolves(rawResponse);
+				const result = await uploadTask.upload();
+				assert.isDefined(result);
+				assert.instanceOf(result, UploadResult);
+				assert.equal(result["location"], location);
+				assert.isTrue(isProgressCompletedCalled);
+				assert.isFalse(isProgressReportCalled);
+				assert.isFalse(isProgressFailureCalled);
+			});
+
+			it("Test without progress callback", async () => {
+				const uploadTask = new LargeFileUploadTask(getClient(), fileObj, uploadSession, options);
+				const status201 = {
+					status: 201,
+					statusText: "OK",
+					headers: {
+						"Content-Type": "application/json",
+						location,
+					},
+				};
+
+				const rawResponse = new Response(JSON.stringify(body), status201);
+				const moq = sinon.mock(uploadTask);
+				moq.expects("uploadSliceGetRawResponse").resolves(rawResponse);
+				const result = await uploadTask.upload();
+				assert.isDefined(result);
+				assert.instanceOf(result, UploadResult);
+				assert.equal(result["location"], location);
+				const responseBody = result["responseBody"];
+				assert.isDefined(responseBody);
+				assert.equal(responseBody["id"], "TEST_ID");
+			});
 		});
 
-		it("Should return a Upload Result object after a completed task with 200 status and id", async () => {
-			const location = "TEST_URL";
-			const body = {
-				id: "TEST_ID",
-			};
-			const uploadTask = new LargeFileUploadTask(getClient(), fileObj, uploadSession, options);
-			const status200 = {
-				status: 200,
-				stautsText: "OK",
-				headers: {
-					"Content-Type": "application/json",
-					location,
-				},
-			};
-			const rawResponse = new Response(JSON.stringify(body), status200);
-
-			const moq = sinon.mock(uploadTask);
-			moq.expects("uploadSliceGetRawResponse").resolves(rawResponse);
-			const result = await uploadTask.upload();
-			assert.isDefined(result);
-			assert.instanceOf(result, UploadResult);
-			assert.equal(result["location"], location);
-			const responseBody = result["responseBody"];
-			assert.isDefined(responseBody);
-			assert.equal(responseBody["id"], "TEST_ID");
+		describe("Should return a Upload Result object after a completed task with 200 status and body", () => {
+			it("Test with completed progress callback", async () => {
```

Comment:
naming convention:
I did some reading, and it looks like `describe()` blocks take objects or function names and `it()` blocks take an expected action, e.g. `should call completed callback`. I think existing tests from this file are also following this convention.

Some reference documents:
https://sinonjs.org/releases/v9.2.4/stubs/
https://www.sitepoint.com/sinon-tutorial-javascript-testing-mocks-spies-stubs/

Reply:
Done